src/test_util/components.rs

```diff
@@ -65,12 +65,36 @@ pub const HTTP_SINK_TAGS: [&str; 2] = ["endpoint", "protocol"];
 /// The standard set of tags for all `AWS`-based sinks.
 pub const AWS_SINK_TAGS: [&str; 2] = ["protocol", "region"];
 
-/// The list of source sender buffer metrics that must be emitted.
-const SOURCE_SENDER_BUFFER_METRICS: [&str; 2] = [
-    "source_buffer_utilization",
-    "source_buffer_utilization_level",
+/// The set of suffixes that define the source/transform buffer metric family.
+const BUFFER_METRIC_SUFFIXES: [&str; 3] = [
+    // While hypothetically possible, the `max_byte_size` metric is never actually emitted, because
+    // both sources and transforms limit their buffers by event count. If we ever allow
+    // configuration by byte size, we will need to account for this in these tests.
+    "max_event_size",
+    "utilization",
+    "utilization_level",
 ];
 
+/// Buffer metric requirements shared between sources and transforms.
+#[derive(Clone, Copy)]
+struct BufferMetricRequirement {
+    prefix: &'static str,
+    suffixes: &'static [&'static str],
+    required_tags: &'static [&'static str],
+}
+
+const SOURCE_BUFFER_METRIC_REQUIREMENT: BufferMetricRequirement = BufferMetricRequirement {
+    prefix: "source_buffer_",
+    suffixes: &BUFFER_METRIC_SUFFIXES,
+    required_tags: &["output"],
+};
+
+const TRANSFORM_BUFFER_METRIC_REQUIREMENT: BufferMetricRequirement = BufferMetricRequirement {
+    prefix: "transform_buffer_",
+    suffixes: &BUFFER_METRIC_SUFFIXES,
+    required_tags: &[],
+};
```

Comment:
If I'm understanding this correctly only one of these two are going to exist but never both for the same component right?

Reply:
Correct. All sources should have the former, and all transforms the latter, and never both.